File: ModoMercado/Core/Domain/Models/Produto.swift
import Foundation

struct Produto: Identifiable, Equatable, Codable {
    let id: UUID
    var nome: String
    var categoria: String
    var unidade: String
    var marca: String?
    var quantidade: Double?
    var observacao: String?
    let criadoEm: Date
    var atualizadoEm: Date

    init(
        id: UUID = UUID(),
        nome: String,
        categoria: String,
        unidade: String,
        marca: String? = nil,
        quantidade: Double? = nil,
        observacao: String? = nil,
        criadoEm: Date = Date(),
        atualizadoEm: Date = Date()
    ) {
        self.id = id
        self.nome = nome
        self.categoria = categoria
        self.unidade = unidade
        self.marca = marca
        self.quantidade = quantidade
        self.observacao = observacao
        self.criadoEm = criadoEm
        self.atualizadoEm = atualizadoEm
    }
}

File: modo_mercado/lib/core/domain/models/produto.dart
import 'package:equatable/equatable.dart';

class Produto extends Equatable {
  final String id;
  final String nome;
  final String categoria;
  final String unidade;
  final String? marca;
  final double? quantidade;
  final String? observacao;
  final DateTime criadoEm;
  final DateTime atualizadoEm;

  const Produto({
    required this.id,
    required this.nome,
    required this.categoria,
    required this.unidade,
    this.marca,
    this.quantidade,
    this.observacao,
    required this.criadoEm,
    required this.atualizadoEm,
  });

  Produto copyWith({
    String? id,
    String? nome,
    String? categoria,
    String? unidade,
    String? marca,
    double? quantidade,
    String? observacao,
    DateTime? criadoEm,
    DateTime? atualizadoEm,
  }) {
    return Produto(
      id: id ?? this.id,
      nome: nome ?? this.nome,
      categoria: categoria ?? this.categoria,
      unidade: unidade ?? this.unidade,
      marca: marca ?? this.marca,
      quantidade: quantidade ?? this.quantidade,
      observacao: observacao ?? this.observacao,
      criadoEm: criadoEm ?? this.criadoEm,
      atualizadoEm: atualizadoEm ?? this.atualizadoEm,
    );
  }

  @override
  List<Object?> get props => [
        id, nome, categoria, unidade, marca,
        quantidade, observacao, criadoEm, atualizadoEm,
      ];
}
